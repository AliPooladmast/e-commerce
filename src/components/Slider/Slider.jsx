import { ArrowLeftOutlined, ArrowRightOutlined } from "@material-ui/icons";
import React from "react";
import style from "./Slider.module.scss";
import fashionSet from "../../assests/images/fashion-set.png";

const Slider = () => {
  return (
    <div className={style.Container}>
      <div className={`${style.Arrow} ${style.Left}`}>
        <ArrowLeftOutlined />
      </div>

      <div className={style.Wrapper}>
        <div className={style.Slide}>
          <div className={style.ImgContainer}>
            <img src={fashionSet} alt="a fashion set" />
          </div>
          <div className={style.InfoContainer}></div>
        </div>
      </div>

      <div className={`${style.Arrow} ${style.Right}`}>
        <ArrowRightOutlined />
      </div>
    </div>
  );
};

export default Slider;
