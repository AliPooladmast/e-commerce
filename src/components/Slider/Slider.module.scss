.Container {
  width: 100%;
  height: 75vh;
  display: flex;
  position: relative;
  overflow: hidden;
  margin-top: 10px;

  @media only screen and (max-width: 600px) {
    display: none;
  }
}

.Arrow {
  width: 50px;
  height: 50px;
  background-color: rgb(255, 247, 247);
  border-radius: 50%;
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  top: 0;
  bottom: 0;
  margin: auto;
  cursor: pointer;
  opacity: 0.5;
  z-index: 2;

  &--Left {
    @extend .Arrow;
    left: 10px;
  }

  &--Right {
    @extend .Arrow;
    right: 10px;
  }
}

.Wrapper {
  height: 100%;
  display: flex;
  transition: all 1.5s ease;
}

.Slide {
  height: 100%;
  width: 100vw;
  display: flex;
  align-items: center;
}

.ImgContainer {
  height: 100%;
  flex: 1;
}

.Image {
  height: 95%;
}

.InfoContainer {
  flex: 1;
}

.Title {
  font-size: 50px;
}

.Desc {
  margin: 50px 0;
  font-size: 20px;
  font-weight: 500;
  letter-spacing: 3px;
}

.Button {
  padding: 10px;
  font-size: 20px;
  background-color: transparent;
  cursor: pointer;
}
