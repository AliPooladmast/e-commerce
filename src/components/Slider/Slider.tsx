import { ArrowLeftOutlined, ArrowRightOutlined } from "@mui/icons-material";
import React, { useState } from "react";
import style from "./Slider.module.scss";
import { sliderItems } from "../../data";
import { Link } from "react-router-dom";

interface ISlide {
  id: number;
  img: string;
  imgMobile: string;
  title: string;
  desc: string;
}

const Slider = () => {
  const [slideIndex, setSlideIndex] = useState<number>(0);

  const handleClick = (direction: string) => {
    if (direction === "left")
      setSlideIndex(slideIndex > 0 ? slideIndex - 1 : 3);
    else setSlideIndex(slideIndex < 3 ? slideIndex + 1 : 0);
  };

  return (
    <div className={style.Container}>
      <div className={style["Arrow--Left"]} onClick={() => handleClick("left")}>
        <ArrowLeftOutlined />
      </div>

      <div
        data-testid="slider-wrapper"
        className={style.Wrapper}
        style={{ transform: `translateX(${slideIndex * -100}vw)` }}
      >
        {sliderItems.map((item: ISlide) => (
          <div className={style.Slide} key={item.id}>
            <div className={style.ImgContainer}>
              <img
                className={style.Image}
                src={item.img}
                srcSet={`${item.imgMobile} 300w, ${item.img} 700w`}
                sizes="(max-width: 850px) 300px, 700px"
                alt="slider"
              />
            </div>

            <div className={style.InfoContainer}>
              <h1 className={style.Title}>{item.title}</h1>
              <p className={style.Desc}>{item.desc}</p>
              <Link to="products">
                <button className={style.Button}>SHOP NOW</button>
              </Link>
            </div>
          </div>
        ))}
      </div>

      <div
        data-testid="arrow-right"
        className={style["Arrow--Right"]}
        onClick={() => handleClick("right")}
      >
        <ArrowRightOutlined />
      </div>
    </div>
  );
};

export default Slider;
