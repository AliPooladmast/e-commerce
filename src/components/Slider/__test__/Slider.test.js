const { render, screen, waitFor } = require("@testing-library/react");
const { BrowserRouter } = require("react-router-dom");
import Slider from "../Slider";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";

it("check if sliding works", async () => {
  render(
    <BrowserRouter>
      <Slider />
    </BrowserRouter>
  );

  const user = userEvent.setup();
  user.click(screen.getByTestId("arrow-right"));

  await waitFor(() => {
    expect(screen.getByTestId("slider-wrapper")).toHaveStyle(
      "transform: translateX(-100vw)"
    );
  });
});

it("check if shop link is correct", () => {
  render(
    <BrowserRouter>
      <Slider />
    </BrowserRouter>
  );

  expect(
    screen
      .getAllByTestId("shop-link")
      .every((item) => item.href.includes("products"))
  ).toBeTruthy();
});
