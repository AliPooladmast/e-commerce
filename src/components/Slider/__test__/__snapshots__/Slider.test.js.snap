// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="Container"
>
  <div
    className="Arrow--Left"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowLeftOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m14 7-5 5 5 5V7z"
      />
    </svg>
  </div>
  <div
    className="Wrapper"
    data-testid="slider-wrapper"
    style={
      Object {
        "transform": "translateX(0vw)",
      }
    }
  >
    <div
      className="Slide"
    >
      <div
        className="ImgContainer"
      >
        <img
          alt="slider"
          className="Image"
          sizes="(max-width: 850px) 300px, 700px"
          src="./assets/images/fashion-set.jpg"
          srcSet="./assets/images/fashion-set-mobile.jpg 300w, ./assets/images/fashion-set.jpg 700w"
        />
      </div>
      <div
        className="InfoContainer"
      >
        <h1
          className="Title"
        >
          SEASONAL SALE
        </h1>
        <p
          className="Desc"
        >
          NEW PRODUCTS WOULD BE DISCOUNTED UP TO 30%
        </p>
        <a
          data-testid="shop-link"
          href="/products"
          onClick={[Function]}
        >
          <button
            className="Button"
          >
            SHOP NOW
          </button>
        </a>
      </div>
    </div>
    <div
      className="Slide"
    >
      <div
        className="ImgContainer"
      >
        <img
          alt="slider"
          className="Image"
          sizes="(max-width: 850px) 300px, 700px"
          src="./assets/images/girl-sundress.jpg"
          srcSet="./assets/images/girl-sundress-mobile.jpg 300w, ./assets/images/girl-sundress.jpg 700w"
        />
      </div>
      <div
        className="InfoContainer"
      >
        <h1
          className="Title"
        >
          SPECTIAL COLLECTION
        </h1>
        <p
          className="Desc"
        >
          NEW PRODUCTS WOULD BE DISCOUNTED UP TO 30%
        </p>
        <a
          data-testid="shop-link"
          href="/products"
          onClick={[Function]}
        >
          <button
            className="Button"
          >
            SHOP NOW
          </button>
        </a>
      </div>
    </div>
    <div
      className="Slide"
    >
      <div
        className="ImgContainer"
      >
        <img
          alt="slider"
          className="Image"
          sizes="(max-width: 850px) 300px, 700px"
          src="./assets/images/men-coat.jpg"
          srcSet="./assets/images/men-coat-mobile.jpg 300w, ./assets/images/men-coat.jpg 700w"
        />
      </div>
      <div
        className="InfoContainer"
      >
        <h1
          className="Title"
        >
          GIFT FOR FANS
        </h1>
        <p
          className="Desc"
        >
          NEW PRODUCTS WOULD BE DISCOUNTED UP TO 30%
        </p>
        <a
          data-testid="shop-link"
          href="/products"
          onClick={[Function]}
        >
          <button
            className="Button"
          >
            SHOP NOW
          </button>
        </a>
      </div>
    </div>
    <div
      className="Slide"
    >
      <div
        className="ImgContainer"
      >
        <img
          alt="slider"
          className="Image"
          sizes="(max-width: 850px) 300px, 700px"
          src="./assets/images/girl-sunglass.jpg"
          srcSet="./assets/images/girl-sunglass-mobile.jpg 300w, ./assets/images/girl-sunglass.jpg 700w"
        />
      </div>
      <div
        className="InfoContainer"
      >
        <h1
          className="Title"
        >
          FASHION PORDUCTS
        </h1>
        <p
          className="Desc"
        >
          NEW PRODUCTS WOULD BE DISCOUNTED UP TO 30%
        </p>
        <a
          data-testid="shop-link"
          href="/products"
          onClick={[Function]}
        >
          <button
            className="Button"
          >
            SHOP NOW
          </button>
        </a>
      </div>
    </div>
  </div>
  <div
    className="Arrow--Right"
    data-testid="arrow-right"
    onClick={[Function]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowRightOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m10 17 5-5-5-5v10z"
      />
    </svg>
  </div>
</div>
`;
